src/reanimated2/helperTypes.ts

```diff
@@ -23,24 +23,41 @@ import type { ReanimatedKeyframe } from './layoutReanimation/animationBuilder/Ke
 import type { SharedTransition } from './layoutReanimation/sharedTransitions';
 import type { DependencyList } from './hook/commonTypes';
 
+/**
+ * @deprecated
+ */
 export type Adaptable<T> =
   | T
   | ReadonlyArray<T | ReadonlyArray<T>>
   | SharedValue<T>;
 
+/**
+ * @deprecated
+ */
 export type AdaptTransforms<T> = {
   [P in keyof T]: Adaptable<T[P]>;
 };
 
-type TransformsStyle = Pick<RNTransformsStyle, 'transform'>;
+type TransformStyle = RNTransformsStyle['transform'];
 
-export type TransformStyleTypes = TransformsStyle['transform'] extends
-  | readonly (infer T)[]
-  | string
-  | undefined
-  ? T
+export type ExtractArrayType<Arr> = Arr extends readonly (infer Element)[]
+  ? Element
   : never;
-export type AnimatedTransform = AdaptTransforms<TransformStyleTypes>[];
+
+/**
+ * @deprecated Please use `TransformArrayType` type instead.
+ */
+export type TransformStyleTypes = ExtractArrayType<TransformStyle>;
+
+export type TransformArrayType = TransformStyleTypes;
+
+// Note: why is `readonly` here? For safety.
+// In TS ReadonlyArray is a supertype of Array,
+// therefore Array can be assigned to ReadonlyArray but
+// ReadonlyArray cannot be assigned to Array.
+export type AnimatedTransform =
```

Comment:
The name of `AnimatedTransform` should be analogous to `TransformStyle` but currently it isn't.